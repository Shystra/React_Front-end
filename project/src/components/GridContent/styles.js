import styled, { css } from 'styled-components';


export const GridContent = styled.div`
    ${({ theme }) => css`
    
    
    `}
`;