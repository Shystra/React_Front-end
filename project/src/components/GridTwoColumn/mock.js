export default {

    title: 'GridTwoColumn',
    text: `Lorem, ipsum dolor sit amet consectetur adipisicing elit.
         Deleniti iusto commodi, illo eligendi laudantium consequuntur,
          nobis facere adipisci, exercitationem inventore ipsum unde odit dolore?
          Nisi nobis ratione praesentium perspiciatis porro`,
    srcImg: 'test2.png',
};