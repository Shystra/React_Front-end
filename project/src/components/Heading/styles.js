import styled, { css } from 'styled-components';



const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};`,

  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};`,

  

  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    
  `,

  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
    ${mediaFont(theme)};`,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.huge};
    background: purple;
    
  }
`;


const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;



export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;