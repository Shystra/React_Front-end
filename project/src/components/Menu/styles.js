import styled, {css} from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css`
        position: fixed;
        z-index: 5;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        border-bottom: ${theme.colors.mediumGray}; 
    
    `}
`;

export const MenuContainer = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;

        @media ${theme.media.lteMedium} {
            display: block;
            text-align: center;
            /* padding: ${theme.spacings.xxlarge}; */
        }
    
    `}
`;
export const Button = styled.button`
    ${({ theme }) => css`
        
    
    `}
`;
