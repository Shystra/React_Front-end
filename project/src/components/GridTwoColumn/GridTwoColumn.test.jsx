import { screen } from '@testing-library/react';
import { GridTwoColumn } from '.'
import { renderTheme } from '../styles/render_theme';
import mock from '.';


describe('<GridTwoColumn />', () => {
    it('should render two column grid', () => {
        const { container } = renderTheme(<GridTwoColumn {...mock} />);
        expect(container).toMatchSnapshot();
    });
});