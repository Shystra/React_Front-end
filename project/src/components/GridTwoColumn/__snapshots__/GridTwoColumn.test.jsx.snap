// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render two column grid 1`] = `
.c5 {
  color: #faebd7;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

.c7 {
  width: 100%;
  border-radius: 10px;
  box-shadow: 0 0 10px rgba(0,0,0,0.3);
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
}

.c0 {
  background: #000000;
  color: #faebd7;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          />
          <p
            class="c6"
          />
        </div>
        <div
          class=""
        >
          <img
            class="c7"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
